qiskit/transpiler/preset_passmanagers/level3.py

```diff
@@ -107,6 +107,8 @@ def level_3_pass_manager(pass_manager_config: PassManagerConfig) -> StagedPassMa
         "scheduling", scheduling_method, pass_manager_config, optimization_level=3
     )
 
+    post_optimization = common.generate_post_op_passmanager()
+
```

Comment:
I think we ought to include the layout finalisation in every optimisation level, even if by our presets it won't have any work to do.  It just helps catch the case of somebody using (say) an O1 pass manager but injecting `ElideSwaps` into pre-op, which is a totally valid use of `StagedPassManager`.

Since we don't have `final_permutation` and its fixing of the update responsibilities to be local to the modifying pass, I think we need to treat `FinalizeLayouts` as a required part of the transpiler pipeline at all optimisation levels, or shift the responsibility of `FinalizeLayouts` into `PassManager._passmanager_backend` (https://github.com/Qiskit/qiskit/pull/9523#issuecomment-2089363405).

Reply:
Since #9523 moved the responsibility of `FinalizeLayouts` into `PassManager._passmanager_backend`, now we don't need the post-op stage at all, rather than needing it everywhere.

Reply:
yes, this is much-much cleaner than having a separate `FinalizeLayouts` pass.